packages/widgetbook/lib/src/fields/color_field/color_field.dart

```diff
@@ -0,0 +1,223 @@
+import 'package:flutter/material.dart';
+
+import '../field.dart';
+import '../field_codec.dart';
+import '../field_type.dart';
+import 'color_space.dart';
+import 'hex_color_text_fields.dart';
+import 'hsl_color_text_fields.dart';
+import 'rgba_color_text_fields.dart';
+
+export 'color_space.dart';
+
+/// [Field] that builds [ColorsFieldWidget] for [Color] values.
+///
+/// The [ColorField] uses the [ColorFieldCodec] to convert the [Color] value to a
+/// hex string and vice versa.
+///
+/// The [ColorField] uses the [ColorSpace] to determine which format the color is.
+class ColorField extends Field<Color> {
+  ColorField({
+    required super.name,
+    super.initialValue = defaultColor,
+    this.initialColorSpace = ColorSpace.hex,
+    super.onChanged,
+  }) : super(
+          type: FieldType.color,
+          codec: FieldCodec(
+            toParam: (color) => color.value.toRadixString(16),
+            toValue: (param) {
+              if (param == null) return null;
+              if (param == '0') return Colors.transparent;
+              if (param.length == 6) param = 'ff' + param;
+              return Color(
+                int.parse(
+                  param,
+                  radix: 16,
+                ),
+              );
+            },
+          ),
+        );
+
+  ColorSpace initialColorSpace;
+
+  static const defaultColor = Colors.white;
+
+  @override
+  Widget toWidget(BuildContext context, String group, Color? value) {
+    return ColorsFieldWidget(
+      name: name,
+      colorSpace: initialColorSpace,
+      value: value,
+      initialValue: initialValue,
+      defaultColor: defaultColor,
+      paramValue: codec.toParam(value ?? initialValue ?? defaultColor),
+      onChanged: (value) {
+        updateField(
+          context,
+          group,
+          value,
+        );
+      },
+    );
+  }
+}
+
+class ColorsFieldWidget extends StatefulWidget {
+  const ColorsFieldWidget({
+    required this.name,
+    required this.colorSpace,
+    required this.paramValue,
+    required this.initialValue,
+    required this.value,
+    required this.defaultColor,
+    required this.onChanged,
+    super.key,
+  });
+
+  final ColorSpace colorSpace;
+  final String name;
+  final Color? value;
+  final Color? initialValue;
+  final Color defaultColor;
+  final String paramValue;
+  final ValueChanged<Color> onChanged;
+
+  @override
+  State<ColorsFieldWidget> createState() => _ColorsFieldWidgetState();
+}
+
+class _ColorsFieldWidgetState extends State<ColorsFieldWidget> {
+  late ColorSpace initialColorSpace;
+  late dynamic colorValue;
+
+  @override
+  void initState() {
+    super.initState();
+    initialColorSpace = widget.colorSpace;
+    colorValue = getValueByColorSpace(
+      colorSpace: initialColorSpace,
+      value: widget.paramValue,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
The `ColorPicker` widget will be responsible for the alpha value, and it will no longer be the responsibly of the sub-widgets.

Here's how the design should look like:

![image](https://github.com/widgetbook/widgetbook/assets/41103290/8aa20e61-5c11-41cb-a91b-cbebf8783346)

The second row will change depending on the ColorSpace picked.